src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -290,5 +294,26 @@ static async ValueTask DisposeAsync(object o)
                 }
             }
         }
+
+        private string DebuggerToString()
+        {
+            return $@"ApplicationName = ""{_hostEnvironment.ApplicationName}"", IsRunning = {(IsRunning ? "true" : "false")}";
+        }
+
+        // Host is running if the app has been started and hasn't been stopped.
+        private bool IsRunning => _applicationLifetime.ApplicationStarted.IsCancellationRequested && !_applicationLifetime.ApplicationStopped.IsCancellationRequested;
+
+        internal sealed class HostDebugView(Host host)
+        {
+            private readonly Host _host = host;
```

Comment:
This isn't necessary with primary ctors... all the references to `_host` below can instead just be to `host` and then you can delete this field.

Reply:
I know, but using the primary ctor argument everywhere _feels_ too magical to me. I like the idea that state in methods comes from fields and method arguments. Maybe I'm too conservative.

Would you prefer to remove the field and use the primary argument?

Reply:
> Would you prefer to remove the field and use the primary argument?

Yes... if we're going to use primary ctors, let's use them as they're intended to be used :)